src/coreclr/src/debug/daccess/dacdbiimpl.cpp

```diff
@@ -3570,6 +3571,137 @@ HRESULT DacDbiInterfaceImpl::GetDelegateTargetObject(
+{
+    // The JIT Code heap will always have
+    EEJitManager *pEM = ExecutionManager::GetEEJitManager();
+    _ASSERT(pEM != NULL && pEM->m_pCodeHeap.IsValid());
+
+    PTR_HeapList pHeapList = pEM->m_pCodeHeap;
+    while (pHeapList != NULL)
+    {
+        CodeHeap *pHeap = pHeapList->pHeap;
+        DacpJitCodeHeapInfo jitCodeHeapInfo = DACGetHeapInfoForCodeHeap(pHeap);
+
+        switch (jitCodeHeapInfo.codeHeapType)
+        {
+            case CODEHEAP_LOADER:
+            {
+                TADDR targetLoaderHeap = CLRDATA_ADDRESS_TO_TADDR(jitCodeHeapInfo.LoaderHeap);
+                LOG((LF_CORDB, LL_INFO10000,
+                    "DDBII::EMRFJCH: LoaderCodeHeap 0x%x with LoaderHeap at 0x%x\n",
+                    PTR_HOST_TO_TADDR(pHeap), targetLoaderHeap));
+                PTR_ExplicitControlLoaderHeap pLoaderHeap = PTR_ExplicitControlLoaderHeap(targetLoaderHeap);
+                pLoaderHeap->EnumPageRegions(TrackMemoryRangeHelper, TO_TADDR(rangeAcummulator));
+                break;
+            }
+
+            case CODEHEAP_HOST:
+            {
+                LOG((LF_CORDB, LL_INFO10000,
+                    "DDBII::EMRFJCH: HostCodeHeap 0x%x\n",
+                    PTR_HOST_TO_TADDR(pHeap)));
+                rangeAcummulator->Push({
+                    CLRDATA_ADDRESS_TO_TADDR(jitCodeHeapInfo.HostData.baseAddr),
+                    CLRDATA_ADDRESS_TO_TADDR(jitCodeHeapInfo.HostData.currentAddr)
+                });
+                break;
+            }
+
+            default:
+            {
+                LOG((LF_CORDB, LL_INFO10000, "DDBII::EMRFJCH: unknown heap type at 0x%x\n\n", pHeap));
+                break;
+            }
+        }
+
+        pHeapList = pHeapList->GetNext();
+    }
+}
+
+HRESULT DacDbiInterfaceImpl::GetLoaderHeapMemoryRanges(DacDbiArrayList<COR_MEMORY_RANGE> *pRanges)
+{
+    LOG((LF_CORDB, LL_INFO10000, "DDBII::GLHMR\n"));
+    DD_ENTER_MAY_THROW;
+
+    HRESULT hr = S_OK;
+
+    EX_TRY
+    {
+        CQuickArrayList<COR_MEMORY_RANGE> memoryRanges;
+
+        // Anything that's loaded in the SystemDomain or into the main AppDomain's default context in .NET Core
+        // and after uses only one global allocator. Enumerating that one is enough for most purposes.
+        PTR_LoaderAllocator pGlobalAllocator = SystemDomain::System()->GetLoaderAllocator();
+        _ASSERTE(pGlobalAllocator);
+        EnumerateMemRangesForLoaderAllocator(pGlobalAllocator, &memoryRanges);
+        // TODO: This doesn't consider any uses of AssemblyLoadingContexts (Unloadable or not). Each context has
+        // it's own LoaderAllocator, but there's no easy way of getting a hand at them other than going through
+        // the heap, getting a managed LoaderAllocators, from there getting a Scout, and from there getting a native
+        // pointer to the LoaderAllocator tos enumerate.
+
+        EnumerateMemRangesForJitCodeHeaps(&memoryRanges);
+
+        // TODO: Add R2R Jit heaps
```

Comment:
How much extra work was R2R? I would have thought it would be fairly trivial

Reply:
I left this in the original exploration to find if there's any work to do, as SOS and the DBI never touch the `ReadyToRunJitManager`. For the `EEJitManager`, I could see the heaps stored in `m_pCodeHeap` in places like `allocCodeRaw`. I never found an equivalent allocator anywhere in the code paths there nonetheless.